### PR TITLE
[MM-56033]Added unit tests for UpdateUserCustomStatus and RemoveUserCustomStatu…

### PR DESCRIPTION
**Pull request by fume4mattermost**
…s functions

#### Summary

Added missing unit tests for the following functions in [mattermost/server/channels/api4/status.go at master · mattermost/mattermost](https://github.com/mattermost/mattermost/blob/master/server/channels/api4/status.go) :
updateUserCustomStatus
getUserStatusesByIds
removeUserCustomStatus

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-56033

#### Release Note

```release-note

```

### CODE REVIEW

File: server/channels/api4/status_test.go [hunk 1]
```diff
@@ -243,3 +245,175 @@ func TestUpdateUserStatus(t *testing.T) {
...
+        require.NoError(t, err)
+        CheckOKStatus(t, resp)
+
+        user, _, err := client.GetUser(context.Background(), th.BasicUser.Id, "")
+        require.NoError(t, err)
+        customStatusJSON := user.Props["customStatus"]
+        require.NotEmpty(t, customStatusJSON)
+        var customStatus map[string]interface{}
+        err = json.Unmarshal([]byte(customStatusJSON), &customStatus)
+        require.NoError(t, err)
```
Comment (agnivade) [inline server/channels/api4/status_test.go:269]: Just use `user.GetCustomStatus()`

File: server/channels/api4/status_test.go [hunk 2]
```diff
@@ -243,3 +245,175 @@ func TestUpdateUserStatus(t *testing.T) {
...
+		th.App.UpdateConfig(func(cfg *model.Config) { *cfg.TeamSettings.EnableCustomUserStatuses = false })
+		defer th.App.UpdateConfig(func(cfg *model.Config) { *cfg.TeamSettings.EnableCustomUserStatuses = true })
+
+		toUpdateCustomStatus := &model.CustomStatus{
+			Emoji: "palm_tree",
+			Text:  "My custom status",
+		}
+		_, resp, err := client.UpdateUserCustomStatus(context.Background(), th.BasicUser.Id, toUpdateCustomStatus)
+		require.Error(t, err)
+		CheckNotImplementedStatus(t, resp)
```
Comment (agnivade) [inline server/channels/api4/status_test.go:314]: Also check that the error id is `api.custom_status.disabled`

File: server/channels/api4/status_test.go [hunk 3]
```diff
@@ -243,3 +245,175 @@ func TestUpdateUserStatus(t *testing.T) {
...
+        user, _, err := client.GetUser(context.Background(), th.BasicUser.Id, "")
+        require.NoError(t, err)
+        customStatusJSON := user.Props["customStatus"]
+        require.NotEmpty(t, customStatusJSON)
+        var customStatus map[string]interface{}
+        err = json.Unmarshal([]byte(customStatusJSON), &customStatus)
+        require.NoError(t, err)
+        assert.Equal(t, toUpdateCustomStatus.Emoji, customStatus["emoji"])
+        assert.Equal(t, toUpdateCustomStatus.Text, customStatus["text"])
+        assert.Equal(t, toUpdateCustomStatus.Duration, customStatus["duration"])
```
Comment (agnivade) [inline server/channels/api4/status_test.go:294]: Same as before

File: server/channels/api4/status_test.go [hunk 4]
```diff
@@ -243,3 +245,175 @@ func TestUpdateUserStatus(t *testing.T) {
...
+		_, _, err := client.UpdateUserCustomStatus(context.Background(), th.BasicUser.Id, toUpdateCustomStatus)
+		require.NoError(t, err)
+
+		resp, err := client.RemoveUserCustomStatus(context.Background(), th.BasicUser.Id)
+		require.NoError(t, err)
+		CheckOKStatus(t, resp)
+
+		user, _, err := client.GetUser(context.Background(), th.BasicUser.Id, "")
+		require.NoError(t, err)
+		assert.Empty(t, user.Props["customStatus"])
```
Comment (agnivade) [inline server/channels/api4/status_test.go:371]: Same

**Review by agnivade — COMMENTED** [review]
Great work on adding tests. Left some comments.

**Review by JulienTant — CHANGES_REQUESTED** [review]
Hey @fume4mattermost, thanks for your contribution!

In addition to the comments @agnivade left earlier, it looks like the file has not been formatted (there's a mix of tabs and spaces, the import order are not correct also).

**Review by agnivade — APPROVED** [review]
Nice bot!

**Comment by jwilander** [comment]
@fume4mattermost can you merge latest master into this branch too?

@JulienTant mind taking a look at this PR?

**Review by JulienTant — APPROVED** [review]
Thank you for your contribution @fume4mattermost!

And thank you @jwilander for the reminder, it fell off my radar!

**Comment by agnivade** [comment]
e2e tests passed. Merging.

**Pull request merged**